com.unity.render-pipelines.universal/Runtime/DeferredLights.cs

```diff
@@ -1174,7 +1174,7 @@ void SetupMainLightConstants(CommandBuffer cmd, ref LightData lightData)
             Vector4 lightPos, lightColor, lightAttenuation, lightSpotDir, lightOcclusionChannel;
             UniversalRenderPipeline.InitializeLightConstants_Common(lightData.visibleLights, lightData.mainLightIndex, out lightPos, out lightColor, out lightAttenuation, out lightSpotDir, out lightOcclusionChannel);
 
-            UniversalAdditionalLightData additionalLightData = ForwardLights.GetAdditionalLightData(lightData.visibleLights[lightData.mainLightIndex].light);
+            var additionalLightData = lightData.visibleLights[lightData.mainLightIndex].light.GetUniversalAdditionalLightData();
```

Comment:
I believe this is the only "unsafe" change out of the suggested changes. Before this code would pass if `light == null`. Now it won't.

Reply:
Should we then split this in to 2 or more lines to be sure we're not breaking anything?

`Light light = lightData.visibleLights[lightData.mainLightIndex].light
if (light == null)
    return;
`

Reply:
Shouldn't the null check be performed in the extension method instead and make sure you return an instance to additional light data in case of null?

Reply:
Maybe just get lights and these into an array (matching visible lights) in lightData once and reuse elsewhere. Cache them by hash of light/lightData or similar and only get them when necessary.

Reply:
To clarify, I do not think `Light` would ever be null here. How can we have an index for our main light when the component itself is null.

I am pointing out that all other code would break before if `Light` _was_ null. This is the only place where before it would work if _Light is null_, after it will break with null reference exception. Taking Elvar's suggestion could work.
I did not add a null check to the extension method to keep behavior and error messages identical to the extension method for `GetUniversalAdditionalCameraData()`, we could add a check but we would need to do this to both places.

Reply:
It would be nice to test this with particles lights for instance. Those will yield a null Light iirc.

Reply:
I created a scene with a particle system that uses the lights module using deferred rendering in URP. I used a break point to make sure the code is ran, the component is not null in this case. 
![image](https://user-images.githubusercontent.com/55388940/118651532-f45c1000-b7e5-11eb-871d-635880d70fe9.png)

We could change the code to this - just to be sure:
`            uint lightLayerMask = (uint)LightLayerEnum.LightLayerDefault;
            var light = lightData.visibleLights[lightData.mainLightIndex].light;
            if(light != null)
            {
                var additionalLightData = light.GetUniversalAdditionalLightData();
                lightLayerMask = (uint)additionalLightData.lightLayerMask;
            }`